src/main/java/bio/terra/service/auth/iam/sam/SamIam.java

```diff
@@ -314,6 +315,48 @@ private Map<IamRole, String> syncSnapshotResourcePoliciesInner(
     return policies;
   }
 
+  @Override
+  public Map<IamRole, String> createSnapshotBuilderRequestResource(
+      AuthenticatedUserRequest userReq, UUID snapshotId, UUID snapshotBuilderRequestId)
+      throws InterruptedException {
+    SamRetry.retry(
+        configurationService,
+        () ->
+            createSnapshotBuilderRequestResourceInner(
+                userReq, snapshotId, snapshotBuilderRequestId));
+    return Map.of(IamRole.OWNER, userReq.getEmail());
```

Comment:
Is there a way to eliminate the duplication with line 354 below? Or if not, we could pass the policy setting to  to `createSnapshotBuilderRequestResourceInner()`, so it would only be duplicated within a single method.